### ISSUE TITLE
Will not let me adjust the mmWave sensitivity for the EP1

### ISSUE THREAD

**Issue by briandonna** [RobinBol/io.everythingsmart#9]
The app will not let me change the “mmWave sensitivity”. When I try to change it and then save it, it says “Missing entity mmwave_sensitivity” Even though it has a value in the box (In my case a 6) the default is a 7.

Any Idea what may be the problem and how to fix it?

Here is a Diagnostics Report if that helps in any way: 1f3aee33-bb42-4d38-b954-7a42182f553c

Thank You for your time and help!
Brian

**Comment by TimoSoof** [comment]
The same here, and also the mmwave distance "missing entity mmwave_distance".

This is a brand new EP1-FULL-01
 and using it with Homey.

ESPHome version 2024.5.2
Project version 1.2.3

**Comment by bjoern247** [comment]
Same Problems here:

- missing entity mmwave_distance when changing the distance
- missing entity mmwave_sensitivity when changing the sensitivity

`Failed to load resource: the server responded with a status of 500 (Internal Server Error)
instrument.js:111 Missing entity mmwave_sensitivity: Missing entity mmwave_sensitivity
    at https://my.homey.app/assets/index-e6512f4c.js:159:815405`

ESPHome version
2024.6.1
Project version
1.2.3

**Comment by RobinBol** [comment]
I believe you are running newer firmware than tested with (see readme for compatible firmware versions), and probably some breaking changes have been introduced.

**Comment by bjoern247** [comment]
@RobinBol Yes. I ordered the sensor and you can only install this software and not manually install older software. 
I can't really change that. Except you know some way where I can select something else than beta or stable here:

https://everythingsmarthome.github.io/everything-presence-one/updating.html

**Comment by RobinBol** [comment]
You should be able to pull older firmware from here, not sure about the installation instructions though.

https://github.com/EverythingSmartHome/everything-presence-one/releases

**Comment by briandonna** [comment]
Hi Robin,
 
I have read all the changes and v1.1.8 https://github.com/EverythingSmartHome/everything-presence-one/releases/tag/1.1.8 is labeled as a breaking change, see here; (##Breaking change
I've marked this as a breaking change, but it's purely because if you have "baseline" numbers in your automations for when the light should come on at a certain lux value, this release will change that baseline figure and will need adjusting.).
 
How much of a donation would it cost to get you to update the app? I will pay ASAP.
 
Thank you
Brian

**Comment by briandonna** [comment]
It looks like this is the last vision before the breaking version v1.3.0b; https://github.com/EverythingSmartHome/everything-presence-one/releases/tag/v1.3.0b
 
It has two downloads, Source code(zip), and https://github.com/EverythingSmartHome/everything-presence-one/archive/refs/tags/v1.3.0b.zip Source code(tar.gz)   https://github.com/EverythingSmartHome/everything-presence-one/archive/refs/tags/v1.3.0b.tar.gz
which one do I use and how do I install it manually? I do not know how to install it manually.
 
I still would like to pay you to update the app so we all can use it as intended.
 
If you do not thank you what to take on the task, would you mind if I get one of my programmers I use to web development to get the app up to date? 
 
Thank You
Brian

**Comment by bjoern247** [comment]
I would also like to donate to get this updated.

**Comment by bjoern247** [comment]
By the way, also flows that use "AND"-Cards which check if the mmWave-Alert is still on are failing because of these breaking changes. Sadly it's not just some sensor settings but also flows are unusable. I did not find a way to update to older firmware.

**Comment by briandonna** [comment]
Hi Robin,

It's been 3 weeks now since our last commutation about the app not working properly, your fix was to do a backward Firmware update, but there are multiple problems with this recommended fix, 1 – I don’t see a way from the Everything Smart site to select an older Firmware, 2 – the millimeter wave board he is now using needs to use one of the newer firmware then your last one that you recommend, 3 – we need the bug fixes and the update of the newer firmware’s to take advantage of the options that are now available.

From what I can find there is only one Firmware version v1.1.8 that will break your code and it is stated in this version see below and here: https://github.com/EverythingSmartHome/everything-presence-one/releases/tag/1.1.8

(v1.1.8
This release adds calibration data for the light sensor to drastically improve the readings with the front cover on.

##Breaking change
I've marked this as a breaking change, but it's purely because if you have "baseline" numbers in your automations for when the light should come on at a certain lux value, this release will change that baseline figure and will need adjusting.)

If you would please let us know if you have any plans to get the app updated or if you are abandoning any and all updates. I would like to buy more of this presence sensor but the drivers need to work first or there is no point. 
I and others are willing to donate funds for your time to get this app up to date. 

Please let us know if we will have a functional app in the near future.

Thanks
Brian

**Comment by RobinBol** [comment]
I've submitted an issue with the firmware developer: https://github.com/EverythingSmartHome/everything-presence-one/issues/197

Once I know what breaking changes were introduced I can take a look at fixing it.

**Comment by EverythingSmartHome** [comment]
Happy to help with whatever is needed to get this back up and working properly

**Comment by briandonna** [comment]
Hi Robin,
Do you have an update as to when you think you will have this app fixed? it’s been three weeks now since Lois responded to your questions as what it would take to fix the app and he even offered to help fix it but there’s been no comment back from you. Would you please respond and let us know when you plan to have the app operational.

Thank you, Brian.

**Comment by RobinBol** [comment]
I'm not entirely sure why the entities `mmwave_distance` and `mmwave_sensitivity` are missing on your EP1's with SEN0609.

If I understand correctly from @EverythingSmartHome's [comment](https://github.com/EverythingSmartHome/everything-presence-one/issues/197#issuecomment-2227145607), only additional entities have been added. So the two above should still be available.

I am not really fluent in esphome config, but a quick search for `mmwave_sensitivity` in the [EverythingSmartHome/everything-presence-one](https://github.com/EverythingSmartHome/everything-presence-one) repo gives me hits for SEN0609 related config:
- https://github.com/EverythingSmartHome/everything-presence-one/blob/7314da04d30a07edabcb386f4afedb2dcc0c0bcf/everything-presence-one-st-sen0609.yaml#L368
- https://github.com/EverythingSmartHome/everything-presence-one/blob/7314da04d30a07edabcb386f4afedb2dcc0c0bcf/common/sen0609-common.yaml#L214

This would make me think that the `mmwave_sensitivity` entity and therefore the "mmWave sensitivity" setting in Homey should work as expected.

Is someone able to run my app from this [branch](https://github.com/RobinBol/io.everythingsmart/tree/debug/entity-logging) [using the Homey CLI](https://apps.developer.homey.app/the-basics/getting-started/homey-cli) and share the whole console output after it has connected with your EP1? It should give me data on the available entities.

Make sure to run `npm ci` to install the app's dependencies and run `homey app run -r` if you are on Homey Pro (Early 2023).

**Comment by briandonna** [comment]
As I read it he is using a "new mmwave board" then what was the original mmwave board that is breaking the code.

**Comment by RobinBol** [comment]
> As I read it he is using a "new mmwave board" then what was the original mmwave board that is breaking the code.

Yes, I understood that as well, but still:

> If I understand correctly from @EverythingSmartHome's https://github.com/EverythingSmartHome/everything-presence-one/issues/197#issuecomment-2227145607, only additional entities have been added. So the two above should still be available.

**Comment by briandonna** [comment]
Hi Robin,
So are you saying that you don’t know what is wrong? Because I still cannot make any changes in the settings, if I try it just will not save the changes so there has to be something wrong.

The everything Smart home EP1 manufacturer offered to help, should I reach out to him to reach out to you to offer a way to find a problem and correct it?

**Comment by EverythingSmartHome** [comment]
Can I ask which smart home platform you selected when flashing the EP1?

**Comment by briandonna** [comment]
Home Assistant

**Comment by EverythingSmartHome** [comment]
Got it - the issue is (I think) that the Home Assistant config firmware requires you to press a button called "set sensitivity" in order to actually change the sensitivity which is different to the previous sensor. I'm not sure if Robin could change this on the Homey side to include the button

**Comment by briandonna** [comment]
Hi Robin,

Please see the answer above from Lewis as to what he thinks may be wrong with the app. Please let us know if you thank you can Implement the button that he suggests. 

Thank You.
Brian

**Comment by briandonna** [comment]
Hi Lewis,

The way his app is setup I have to go to your site to get the updated firmware and IP address and then go to o
Homey and add the device.

do you think I can just redo the firmware and go further into the setup until I get to the sensitivity button? Or dose that button need to be in the homey app where the settings are? 

So to ask you about this but Robin is very slow to answer anything.

thank you very much for your help!
Brian

**Comment by EverythingSmartHome** [comment]
You may try flashing the smartthings firmware instead and see if that works or changed anything

**Comment by RobinBol** [comment]
Thank you for your input, @EverythingSmartHome! It looks like supporting the SEN0609 will require creating a separate driver. The key issue is that the settings differ between the SEN0609 and the SEN0395, and since Homey uses statically defined settings, I would need to know which mmWave sensor is being paired upfront in order to provide the correct settings to the user.

As for the required button press to save the settings: Homey doesn't natively support buttons within driver settings. However, I might be able to detect when the user changes any relevant settings and simulate the button press event programmatically.

I'd be open to adding support for the SEN0609 (including handling the necessary button actions), but I would need access to the device for testing and development.

**Comment by briandonna** [comment]
Do you need a device it self or just access to a device like being able to access my device? if the latter is true then how do I give you my credentials?

**Comment by EverythingSmartHome** [comment]
I can give you a device when we get some back in stock sure. In the meantime, it would be great if someone can try flashing the SmartThings firmware instead of the Home Assistant one. We had similar issues with SmartThings so made the adjustments in the SmartThing firmware instead of the driver.

**Comment by briandonna** [comment]
So, just to be sure as far as I know there are only two firmware, I used the home Home Assistant, are you saying I should try the SnartThings? Or flash the Home Assistant again?
 
It's 5:00 A.M here in the U.S. and I am about to leave for work, but if you replay in time I will flash the device when I get home this afternoon.
 
Thanks,
Brian

**Comment by EverythingSmartHome** [comment]
Yes, from the user guide select SmartThings instead of Home Assistant as the smart home platform. Then select the SEN0609 as the sensor and proceed

**Comment by briandonna** [comment]
Roger that! I will do it after I get home this afternoon and let you know how things went.
 
Thank you for all your help!! :)

**Comment by briandonna** [comment]
Hi Lewis,
 
I tried the SmartThings flash but I could not get it to connect to homey after that, so, I flashed to Home Assistant and was able to reconnect to Homey, but I still have the same problem with not being able to change any settings. 
 
I guess we will have to wait until Robin fixes the drivers.

**Comment by briandonna** [comment]
One other thing, I was able to see the settings and make adjustments on my desktop using the SmartThings firmware but did not see that option using the Home Assistant firmware.
 
I did all that early in the morning, so I had to stop testing and prepare to leave for work. I plan to try one more time this afternoon to get the SmartThings firmware to work with Homey. I noticed I had to restart the Homey app to get the EP1 to connect back to Homey after flashing, but I had already flashed it back to the Home Assistant by then. So, I plan to flash the SmartThings firmware one more time and restart the app to see if it will connect using SmartThings.
 
I will let you know how it goes.
